### PR TITLE
TGP:2722: use for comprehension on data store

### PR DESCRIPTION
**Pull request by omotrium**
# Checklist for developers before reviewing
- [x] I have merged in main
- [x] I have assigned the PR to me
- [x] I ran the code with a clean compile test and ensured there are no warnings or deprecations
- [x] I have run `sbt prePR` to run all tests, format and check the code coverage
- [x] I have written unit tests to cover any code that I wrote, where necessary
- [x] I have linked tickets to any TODOs I have written
- [x] I have run the microservices with the latest SM2 config
- [x] I have checked all the Acceptance Criteria pass
- [x] I have run the pipeline journey tests
- [x] I have run the local journey tests
- [x] I have spoken to the QAs about any changes that may need to be made to the journey tests OR no changes to the journey tests are necessary
- [x] I have moved the ticket to Review in Jira
- [x] I have left a message in tgp-prs on slack to alert other Devs that this ticket is ready for review
- [x] I have checked my PR is not in draft mode

# Checklist for reviewer 1
- [ ] I have added an emoji to the relevant message in tgp-prs to show I am reviewing
- [ ] I have checked out the branch
- [ ] I have run the microservices with the latest SM2 config
- [ ] I have checked all the Acceptance Criteria pass
- [ ] I reviewed the code and left comments where necessary
- [ ] I have left a message on the relevant tgp-prs message to show that I have finished reviewing

# Checklist for reviewer 2
- [ ] I have added an emoji to the relevant message in tgp-prs to show I am reviewing
- [ ] I have checked out the branch
- [ ] I have run the microservices with the latest SM2 config
- [ ] I have checked all the Acceptance Criteria pass
- [ ] I reviewed the code and left comments where necessary
- [ ] I have left a message on the relevant tgp-prs message to show that I have finished reviewing

# Checklist for developers before merging
- [ ] The PR has been approved by 2 other developers
- [ ] I have merged in main
- [ ] I have run `sbt preMerge` to run all tests, check formatting and code coverage
- [ ] I have run the microservices with the latest SM2 config
- [ ] I have checked all the Acceptance Criteria pass
- [ ] I have checked that the QAs have merged in any changes to the journey tests
- [ ] I have run the pipeline journey tests
- [ ] I have run the local journey tests

# Reminders for developers after merging
1. Wait for pipeline to pass
2. Move ticket to QA column in Jira
3. Leave message in tgp-qa on slack to alert QAs that this ticket is ready for testing

### CODE REVIEW

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/EmailController.scala [hunk 1]
```diff
@@ -35,11 +35,12 @@ class EmailController @Inject() (
     with Logging {
 
   def getEmail(eori: String): Action[AnyContent] = identify.async { implicit request =>
-    customsDataStoreConnector
-      .getEmail(eori)
-      .map {
-        case Some(email) => Ok(Json.toJson(email))
-        case None        => NotFound
-      }
+    for {
```
Comment (besscerule) [inline app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/EmailController.scala:38]: for comprehension is only needed when there are multiple flatmaps - if it is one flatmap or less i would leave it

**Comment by besscerule** [comment]
I think there is maybe some refactoring you can do in the store records service?

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/GetRecordsController.scala [hunk 2]
```diff
@@ -42,12 +42,15 @@ class GetRecordsController @Inject() (
     pageOpt: Option[Int],
     sizeOpt: Option[Int]
   ): Action[AnyContent] = (identify andThen storeLatest).async {
-    recordsRepository.getCount(eori).flatMap { totalRecords =>
-      recordsRepository.getMany(eori, pageOpt, sizeOpt).map { records =>
-        val getRecordsResponse =
-          GetRecordsResponse(goodsItemRecords = records, buildPagination(sizeOpt, pageOpt, totalRecords))
-        Ok(Json.toJson(getRecordsResponse))
-      }
+    for {
```
Comment (besscerule) [inline app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/GetRecordsController.scala:45]: I'm not sure this change is needed as there is only one flat map.

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/services/StoreRecordsService.scala [hunk 3]
```diff
@@ -42,47 +42,47 @@ class StoreRecordsService @Inject() (
...
-                    recordsSummaryRepository.update(eori, None, None)
-                  }
+    val timeBeforeInitialCall = clock.instant()
+    for {
+      recordsResponse <- routerConnector.getRecords(eori, lastUpdatedDate, Some(startingPage), Some(pageSize))
+      goodsRecords     = recordsResponse.goodsItemRecords
+      totalRecords     = recordsResponse.pagination.totalRecords
 
-                  false
+      result <- if (goodsRecords.isEmpty) {
```
Comment (besscerule) [inline app/uk/gov/hmrc/tradergoodsprofilesdatastore/services/StoreRecordsService.scala:52]: Can we pull this out into a function?

**Review by besscerule — APPROVED** [review]

**Review by pauloncast — APPROVED** [review]

**Pull request merged**